core/src/main/scala/kafka/raft/KafkaFuturePurgatory.scala

```diff
@@ -63,7 +63,7 @@ class DelayedRaftRequest(future: CompletableFuture[Unit], delayMs: Long)
     if (isExpired.get())
       future.completeExceptionally(new TimeoutException("Request timed out in purgatory"))
     else
-      future.complete(())
+      future.complete(null)
```

Comment:
The annoying this is that `KafkaRaftClient` controls the type parameter of `KafkaFuturePurgatory`. I wasn't able to use the `Unit` type as I had hoped because of this, so that just leaves `Void` and the only possible value is `null`.